precompiles/gov/events.go

```diff
@@ -0,0 +1,47 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package gov
+
+import (
+	"bytes"
+	"reflect"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+
+	cmn "github.com/evmos/evmos/v20/precompiles/common"
+)
+
+const (
+	// EventTypeVote defines the event type for the gov VoteMethod transaction.
+	EventTypeVote = "Vote"
+)
+
+// EmitVoteEvent creates a new event emitted on a Vote transaction.
+func (p Precompile) EmitVoteEvent(ctx sdk.Context, stateDB vm.StateDB, voterAddress common.Address, proposalID uint64, option int32) error {
+	// Prepare the event topics
+	event := p.ABI.Events[EventTypeVote]
+	topics := make([]common.Hash, 2)
+
+	// The first topic is always the signature of the event.
+	topics[0] = event.ID
+	topics[1] = common.BytesToHash(voterAddress.Bytes())
```

Comment:
Is there a reason why `voterAddress` is handled this way? In other precompiles we use `cmn.MakeTopic` which uses a different logic for common.Address:

```go
case common.Address:
	copy(topic[common.HashLength-common.AddressLength:], rule[:])
```

Reply:
no particular reason. We can refactor to make it work as you're saying